lib/email_sponsees_extractor.rb

```diff
@@ -0,0 +1,31 @@
+require 'nokogiri'
+
+class EmailSponseesExtractor
+  def execute(email)
+    mail = Mail.read_from_string(email)
+
+    contacts_from_mail(mail).map(&:strip).reject(&:empty?)
+  end
+
+private
+
+  def contacts_from_mail(mail)
+    if !mail.multipart?
+      lines_from_plain_text(mail.body)
+    elsif mail.text_part
+      lines_from_plain_text(mail.text_part)
+    else
+      lines_from_html(mail)
+    end
+  end
+
+  def lines_from_plain_text(message)
+    message.decoded.lines "\n"
+  end
+
+  def lines_from_html(mail)
+    Nokogiri::HTML(mail.html_part.decoded).xpath("//text()").reject(&:cdata?).map do |node|
```

Comment:
In the absence of alternative solutions, I'll trust that this is the cleanest way to achieve this :-)

Reply:
I've moved the style tag check into the xpath, and it looks nicer I'd say.  Thoughts?